editor/store/effects/posts.js

```diff
@@ -0,0 +1,320 @@
+/**
+ * External dependencies
+ */
+import { BEGIN, COMMIT, REVERT } from 'redux-optimist';
+import { pick, includes } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	resetAutosave,
+	resetPost,
+	updatePost,
+	removeNotice,
+	createSuccessNotice,
+	createErrorNotice,
+} from '../actions';
+import {
+	getCurrentPost,
+	getPostEdits,
+	getEditedPostContent,
+	getAutosave,
+	getCurrentPostType,
+	isEditedPostAutosaveable,
+	isEditedPostSaveable,
+	isEditedPostNew,
+	POST_UPDATE_TRANSACTION_ID,
+} from '../selectors';
+import { resolveSelector } from './utils';
+
+/**
+ * Module Constants
+ */
+const SAVE_POST_NOTICE_ID = 'SAVE_POST_NOTICE_ID';
+export const AUTOSAVE_POST_NOTICE_ID = 'AUTOSAVE_POST_NOTICE_ID';
+const TRASH_POST_NOTICE_ID = 'TRASH_POST_NOTICE_ID';
+
+/**
+ * Request Post Update Effect handler
```

Comment:
These JSDoc descriptions really go to reinforce my worry that we don't really have a good understanding of what's intended to take place in these functions.